src/games/strategy/triplea/delegate/BattleDelegateOrdered.java

```diff
@@ -0,0 +1,114 @@
+      }
+    }
+  }
+
+  private int fightAmphib( IBattle amphib ) {
+    int foughtBattleCount = 0;
+    // are there battles that must occur first
+    for( final IBattle seaBattle : m_battleTracker.getDependentOn( amphib ) ) {
+      if( seaBattle instanceof MustFightBattle && seaBattle != null ) {
+        seaBattle.fight( m_bridge );
+        foughtBattleCount++;
+      }
+    }
+    amphib.fight( m_bridge );
+    foughtBattleCount++;
+    
+    return foughtBattleCount;
+  }
+
+  private int processAmphib() {
+    // Fight all amphibious assaults with no retreat option for the attacker and no sea combat - these have no real attacker decisions
+    // Also remove all remaining defenseless fights by fighting them
+    int battleCount = 0;
+    int amphibCount = 0;
+    IBattle lastAmphib = null;
+    for( final Territory t : m_battleTracker.getPendingBattleSites(false) ) {  // Loop through normal combats i.e. not bombing or air raid
+      final IBattle battle = m_battleTracker.getPendingBattle(t, false, BattleType.NORMAL);
+      if( battle instanceof NonFightingBattle && m_battleTracker.getDependentOn(battle).isEmpty() ) {
+        battle.fight( m_bridge );         // Remove non fighting battles by fighting them automatically. Conveniently done here.
+        continue;
+      } else if (!(battle instanceof MustFightBattle) ) {
+        continue;
+      }
+      battleCount++;
+      if( battle.isAmphibious() ) {
+        // If there is no dependent sea battle and no retreat, fight it now 
+        if( m_battleTracker.getDependentOn(battle).isEmpty() && !( (MustFightBattle) battle).canAnyAttackersRetreat() ) {
+          battle.fight( m_bridge );
+        } else {
+          amphibCount++;
+          lastAmphib = battle;      // Otherwise store it to see if it can be fought later i.e. if there's only one
+        }
+      }
+    }
+
+    // Fight amphibious assault if there is one remaining. Fight naval battles in random order if prerequisites first.
+    if( amphibCount == 1 ) {
+      battleCount -= fightAmphib( lastAmphib );   // Subtract off the number of fought battles 
+    }
+    return battleCount;
+  }
+
+  private void fightRemainingBattle() {
+    // This needs another loop just in case the last otherBattleCount was triggered by a sea combat already fought
+    for( final Territory t : m_battleTracker.getPendingBattleSites(false) ) {  // Will only find one
+      final IBattle lastBattle = m_battleTracker.getPendingBattle( t, false, BattleType.NORMAL );
+      if( lastBattle instanceof MustFightBattle ) {
+        lastBattle.fight( m_bridge );
+      }
+    }
+  }
+
+  /**
+   * After the super class .start() method runs, this is called to fight all battles not requiring significant attacker input,
+   * in the order required for Global 1940. Other maps with the same required order can also use it or add it anyway
+   */
+  @Override
+  public void start() {
+    super.start();
+
+    stratBombing();
```

Comment:
Lots of side effects.
OO is about sending messages between objects, we do not see that here.
Functional programming works in terms of inputs and getting outputs back, there is no data being passed here so we do not have that.
That leaves this being  a bit procedural.. The problem there is that the relationships between these methods needs to be understood to modify any one. In that case a change in one spot can cause a series of cascading errors in other modules.

Recommed find ways to split the methods up so that they do low level calculations and return results back.
The main point of this class is to order battles appopriately. Seems like something like:
```
public void start () {
  for(Battle  battle: getAllBattlesInOrder()) {
     battle.fight()
   }
}

private List<Battle> getAll() {
  List<Battle> battles = ..
   battles.addAll(getAmphib());
   battles.addAll(get..());
   return battles;
}
```

Bottom line, even if you don't take the recommendation above, please see if you can decouple the private methods from each other. A good test is if we could swap the ordering of these lines. The reason for the dependency of the specific ordering is not immediately clear, which makes things hard to understand and modify.